### PR TITLE
[WG][APB-7957][APB-7788] making request to DES to retrieve agents det…

### PR DESCRIPTION
**Pull request by gitwojciech**
…ails

### CODE REVIEW

**Comment by r-melvin** [comment]
test this please

File: app/uk/gov/hmrc/agentassurance/controllers/AgentServicesController.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                       )
+(implicit val appConfig: AppConfig, ec: ExecutionContext) extends BackendController(cc) with AuthActions{
+
+  private val strideRoles = Seq(appConfig.manuallyAssuredStrideRole)
+
+  def getAgencyDetails(arn: Arn): Action[AnyContent] = withAffinityGroupAgentOrStride(strideRoles) {  implicit request =>
+    desConnector
+      .getAgentRecord(arn)
+      .map {
+      case _@AgentDetailsDesResponse(utrO, Some(agencyDetails), _) => Ok(Json.toJson(AgentDetailsResponse(agencyDetails, utrO)))
```
Comment (r-melvin) [inline app/uk/gov/hmrc/agentassurance/controllers/AgentServicesController.scala:46]: What does `utrO` mean? Is it optional? If so, `optUtr` would be a little more descriptive

File: app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentassurance.models
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+
+
+case class AgentDetailsResponse(agencyDetails: AgencyDetails, utr: Option[Utr])
```
Comment (r-melvin) [inline app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala:23]: again `optUtr` would help here

File: it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala [hunk 3]
```diff
@@ -195,4 +204,44 @@ trait DesStubs {
        |   "agentReferenceNumber" : "TestARN"
        |}
             """.stripMargin
+
+  def noPersonalDetailsResponseBodyWithValidData(utr: Option[Utr]) =
```
Comment (r-melvin) [inline it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala:208]:
```suggestion
  def noPersonalDetailsResponseBodyWithValidData(optUtr: Option[Utr]) =
```

File: it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala [hunk 4]
```diff
@@ -195,4 +204,44 @@ trait DesStubs {
...
+    s"""
+       |{
+       |   "isAnOrganisation" : true,
+       |   "contactDetails" : {
+       |      "phoneNumber" : "07000000000"
+       |   },
+       |   "isAnAgent" : true,
+       |   "safeId" : "XB0000100101711",
+       |   """.stripMargin ++ utr.map(x =>
```
Comment (r-melvin) [inline it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala:217]:
```suggestion
       |   """.stripMargin ++ optUtr.map(x =>
```

File: it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala [hunk 5]
```diff
@@ -195,4 +204,44 @@ trait DesStubs {
...
+      s""" "uniqueTaxReference": "${x.value}",
```
Comment (r-melvin) [inline it/uk/gov/hmrc/agentassurance/stubs/DesStubs.scala:218]: what's `x`? It should be `utr`

File: app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala (hunk 2 above)
Comment (gitwojciech) [reply app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala:23]: Here I will not change as that object serialise to "utr". I think it looks better in json message

File: app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+object AgentDetailsResponse {
+  implicit val AgentDetailsResponseFormat: OFormat[AgentDetailsResponse] = Json.format[AgentDetailsResponse]
```
Comment (r-melvin) [inline app/uk/gov/hmrc/agentassurance/models/AgentDetailsResponse.scala:26]:
```suggestion
 implicit val writes: OWrites[AgentDetailsResponse] = (
    (JsPath \ "agencyDetails").write[AgencyDetails] and
      (JsPath \ "utr").writeNullable[String]
    )(unlift(AgentDetailsResponse.unapply))
```

Something like that perhaps?

**Review by r-melvin — APPROVED** [review]

**Pull request merged**